library/Faktory/Ent/Batch.hs

```diff
@@ -0,0 +1,129 @@
+{-# LANGUAGE DerivingVia #-}
+
+-- | Support for the @BATCH@ command (Enterprise only)
+--
+-- <https://github.com/contribsys/faktory/wiki/Ent-Batches>
+--
+-- Batches allow multiple Jobs to be enqueued as a group, with a description
+-- (visible in the admin UI) and Jobs attached to run on completion of all Jobs
+-- within the group (always, or only if all were successful).
+--
+-- Usage:
+--
+-- @
+-- -- Build a Job to run at completion of the Batch. Arguments are the same as
+-- -- you would pass to 'perform' the Job.
+-- onComplete <- buildJob mempty producer myJob
+--
+-- 'runBatchT' ('complete' onComplete <> 'description' "My Batch") producer $ do
+--   -- Use 'batchPerform' instead of 'perform'
+--   void $ 'batchPerform' mempty producer myBatchedJob1
+--   void $ 'batchPerform' mempty producer myBatchedJob2
+-- @
+--
+-- __/NOTE__: This module does not support batched Jobs dynamically adding more
+-- Jobs to the Batch. PRs welcome.
+--
+module Faktory.Ent.Batch
+    (
+    -- * Options
+      BatchOptions
+    , description
+    , complete
+    , success
+
+    -- * Running
+    , runBatchT
+    , BatchT
+    , batchPerform
+    )
+where
+
+import Faktory.Prelude
+
+import Control.Monad.Reader
+import Data.Aeson
+import Data.Aeson.Casing
+import Data.ByteString.Lazy as BSL
+import Data.Semigroup (Last(..))
+import Data.Semigroup.Generic
+import Data.Text.Encoding (decodeUtf8, encodeUtf8)
+import Faktory.Client
+import Faktory.Job
+import Faktory.Producer
+import GHC.Generics
+import GHC.Stack
+
+newtype BatchT a = BatchT (ReaderT BatchId IO a)
+  deriving newtype (Functor, Applicative, Monad, MonadIO, MonadReader BatchId)
+
+newtype BatchId = BatchId Text
+  deriving newtype ToJSON
+
+data BatchOptions arg = BatchOptions
+  { boDescription :: Maybe (Last Text)
+  , boSuccess :: Maybe (Last (Job arg))
+  , boComplete :: Maybe (Last (Job arg))
+  }
+  deriving stock Generic
+  deriving Semigroup via GenericSemigroupMonoid (BatchOptions arg)
```

Comment:
Are you avoiding defining the `Monoid` instance? Then you could use `mempty { ... }` in the smart modifiers below

Reply:
Yeah, this was just oversight. Fixed up after.